src/Admin/UserRESTEndpointExtension.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Class UserRESTEndpointExtension
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * UserRESTEndpointExtension class.
+ *
+ * @since 2.2
+ * @internal
+ */
+class UserRESTEndpointExtension implements Service, Registerable, Delayed {
+
+	/**
+	 * User meta key that stores a template mode for which the "Review" panel was dismissed.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE = 'amp_review_panel_dismissed_for_template_mode';
+
+	/**
+	 * Get registration action.
+	 *
+	 * @return string
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Adds hooks.
+	 */
+	public function register() {
+		$this->register_rest_field();
+	}
+
+	/**
+	 * Register REST field for storing a template mode for which the "Review" panel was dismissed.
+	 */
+	public function register_rest_field() {
+		register_rest_field(
+			'user',
+			self::USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE,
+			[
+				'get_callback'    => [ $this, 'get_review_panel_dismissed_for_template_mode' ],
+				'update_callback' => [ $this, 'update_review_panel_dismissed_for_template_mode' ],
```

Comment:
At first I thought this was missing a `permission_callback`, but I see that is only relevant for routes.